chat-server.py

```diff
@@ -0,0 +1,84 @@
+import socket
+import selectors
+
+HEADER_LENGTH = 10
+HOST = '127.0.0.1'
+PORT = 54321
+clients = []  # List for storing client information
+
+
+def accept_connection(sock):
+	"""Function to accept a new connection"""
+
+	global clients
+	conn, addr = sock.accept()
+	print(f'Connected to by {addr}')
+	conn.setblocking(False)
+
+	# Receive first message. This will contain username
+	user_header = conn.recv(HEADER_LENGTH)
+	username_length = int(user_header.decode("utf-8").strip())
+	username = conn.recv(username_length).decode("utf-8")
+
+	# Register socket with sel 
+	events = selectors.EVENT_READ | selectors.EVENT_WRITE
+	data = {"addr": addr, "username": username, "socket": conn}
+	sel.register(conn, events, data=data)
+	clients.append(data)
+
+
+def service_connection(key, mask):
+	"""Function to service an existing connection"""
+
+	global clients
+	sock = key.fileobj
+	data = key.data
+	# Check for read events
+	if mask & selectors.EVENT_READ:
+		# Receive message from socket
+		msg_header = sock.recv(HEADER_LENGTH)
+		if not msg_header:
+			print(f'Closing connection to {data["addr"]}')
+			sel.unregister(sock)
+			sock.close()
+			clients = [client for client in clients if client["addr"]!=data["addr"]]
+			return None
+		msg_length = int(msg_header.decode("utf-8").strip())
```

Comment:
What if the msg_header doesn't contain a number to parse? int() will throw an exception and the server will crash

Reply:
chat-client.py won't allow a blank message to be sent so don't think this should ever happen?

Reply:
It will if the client has been modified by a snake

Reply:
Fixed